.github/workflows/ci.yml

```diff
@@ -94,3 +94,12 @@ jobs:
       - uses: actions/checkout@v2
       - uses: dtolnay/rust-toolchain@clippy
       - run: cargo clippy --workspace --tests -- -Dclippy::all
+
+  clang-tidy:
+    name: Clang Tidy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install clang-tidy
+        run: sudo apt install clang-tidy-10
+      - name: Run clang-tidy
+        run: clang-tidy-10 src/cxx.cc -- -Iinclude
```

Comment:
This only does source file, i.e. `cxx.cc`.

If you want to also clang-tidy the include, you'd have to either set `--header-include='.*'` or set the `HeaderFilterRegex` parameter in the `.clang-tidy` file.

Default value is `HeaderFilterRegex: ''`. You can see that if you do a `clang-tidy--dump-config`.

Alternatively, you can simply ignore headers and run clang-tidy directly on the `include/cxx.h` header file as well.

Reply:
The next thing then would be the question, whether it's worth using a compilation database to get the right compilation settings (from tests or demo builds). Probably not at this stage. The current approach should be good enough for now.